src/pkg/cli/compose/fixup.go

```diff
@@ -71,7 +71,16 @@ func FixupServices(ctx context.Context, provider client.Provider, project *types
 		// Upload the build context, if any; TODO: parallelize
 		if svccfg.Build != nil {
 			// Pack the build context into a tarball and upload
-			url, err := getRemoteBuildContext(ctx, provider, project.Name, svccfg.Name, svccfg.Build, upload)
+			url, err := getRemoteBuildContext(ctx, provider, project.Name, svccfg.Name, svccfg.Build, upload, func(selectPackBuilder string) {
```

Comment:
This is just a convoluted way to add another return value, so please just do that. Either add a 3rd return value, or have it return a `struct` (like `BuildConfig`)